Automattic/jetpack
# [Boost] Correct usage of unlinked=1 param during signup

Only send `unlinked=1` to the signup flow if the current user is not already connected to Jetpack/WordPress.com.

## Proposed changes:
* Include whether the current user is connected in the `ConnectionStatus` object synced to the UI.
* Only include `unlinked=1` if the current user is not connected during signup.

### Other information:

## Jetpack product discussion
p9dueE-6Hf-p2

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:
* Disconnect from Jetpack
* Connect to Boost. It will establish a site-only connection.
* Verify that the `unlinked=1` GET param is added to the signup URL when following the signup flow
* Use the Jetpack plugin to ensure the current user is connected
* Verify that the `unlinked=1` GET param is _not_ added to the signup URL when following the signup flow.
